app/models/concerns/attachmentable.rb

```diff
@@ -52,9 +52,12 @@ def check_image_dimension(attachment)
     return if attachment.blank? || !/image.*/.match?(attachment.content_type) || attachment.queued_for_write[:original].blank?
 
     width, height = FastImage.size(attachment.queued_for_write[:original].path)
-    matrix_limit  = attachment.content_type == 'image/gif' ? GIF_MATRIX_LIMIT : MAX_MATRIX_LIMIT
 
-    raise Mastodon::DimensionsValidationError, "#{width}x#{height} images are not supported" if width.present? && height.present? && (width * height > matrix_limit)
+    if attachment.content_type == 'image/gif'
+      raise Mastodon::DimensionsValidationError, "#{width}x#{height} GIF files are not supported" if width.present? && height.present? && (width * height > GIF_MATRIX_LIMIT)
+    elsif width.present? && height.present? && (width * height > MAX_MATRIX_LIMIT)
```

Comment:
Thoughts on keeping the `if width.present? && height.present?` check outside the `content_type` conditional? Might keep the check about the error raise more narrowly focused on what's different based on type (the matrix comparison).